src/org/zaproxy/zap/extension/api/ApiResponseSet.java

```diff
@@ -29,11 +29,11 @@
 import org.w3c.dom.Text;
 import org.zaproxy.zap.utils.XMLStringUtil;
 
-public class ApiResponseSet extends ApiResponse {
+public class ApiResponseSet<T> extends ApiResponse {
```

Comment:
Is the parameterization really required? The type is not exposed in the public API and internally it only uses Object (i.e. any type).
How about the use of wildcard for the values of the Map?